### PR TITLE
Add an optional GzipHandler to JettyService configurable via WebConfig

### PR DESCRIPTION
**Pull request by tso**

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt [hunk 1]
```diff
@@ -164,6 +167,13 @@ class JettyService @Inject internal constructor(
     server.stopTimeout = 25_000
     ServerConnectionStatistics.addToAllConnectors(server)
 
+    if (webConfig.gzip) {
+      gzipHandler.server = server
+      gzipHandler.minGzipSize = webConfig.minGzipSize
+      gzipHandler.addIncludedMethods("POST")
```
Comment (swankjesse) [inline misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: what does this do?

File: misk/src/test/kotlin/misk/web/GzipTest.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  private fun call(request: Request.Builder): Response {
+    val httpClient = OkHttpClient()
+    val response = httpClient.newCall(request.build()).execute()
+    assertThat(response.code).isEqualTo(200)
+    return response
+  }
+
+  private fun Response.assertGzipEncoding(gzipped: Boolean) {
+    use {
+      assertThat(networkResponse?.headers?.toMap()?.get("Content-Encoding") == "gzip")
```
Comment (swankjesse) [inline misk/src/test/kotlin/misk/web/GzipTest.kt]: you can shrink this to `networkResponse?.headers?.get("Content-Encoding") == "gzip"`

Comment (swankjesse) [reply misk/src/test/kotlin/misk/web/GzipTest.kt]: (otherwise it needs to create a case-insensitive map!)

**Comment by swankjesse** [comment]
Awesome. For follow-up at some point is accepting Gzipped requests.

File: misk/src/main/kotlin/misk/web/jetty/JettyService.kt (hunk 1 above)
Comment (tso) [reply misk/src/main/kotlin/misk/web/jetty/JettyService.kt]: By default the gzip handler only runs on GET requests. This tells it to also run on POSTs

**Pull request merged**